### PR TITLE
[NL] Display extended stat vars as comparison charts

### PR DESCRIPTION
**Pull request by shifucun**
* Remove unused codes: related places, other commented codes
* Use "extended_sv_maps" as how topic branch works. The major change is to find extended svs robustly by inspecting direct siblings or remote siblings.
* Simplify context by using `data_spec` directly

### CODE REVIEW

File: server/lib/nl/nl_variable.py [hunk 1]
```diff
@@ -43,38 +57,82 @@ class VariableStore:
...
-  return result
-
-
-def extend_svs(svs: List[str]):
-  """Extend all svs with siblings svs under the same svg.
+      res.pvs[k] = v
+  return res
+
+
+def parse_svg(svg_dcid: str) -> SVG:
```
Comment (pradh) [inline server/lib/nl/nl_variable.py:76]: please note that SVG DCID does not directly relate to the schema. We should probably set definitions for SVGs too in the cache?

File: server/lib/nl/nl_variable.py [hunk 2]
```diff
@@ -43,38 +57,82 @@ class VariableStore:
...
+    if "-" in part:
+      p, v = part.split("-")
+      p_lower = p.replace(p[0], p[0].lower(), 1)
+      res.pvs[p_lower] = v
+    else:
+      res.p = part.replace(part[0], part[0].lower(), 1)
+  return res
+
+
+def extend_svs(svs: Dict[str, List[str]]):
```
Comment (pradh) [inline server/lib/nl/nl_variable.py:93]: Could you add some comments on the logic here?

File: server/lib/nl/nl_variable.py [hunk 3]
```diff
@@ -43,38 +57,82 @@ class VariableStore:
...
+        for sv_info in item['info'].get('childStatVars', []):
+          curr_sv_obj = parse_sv(sv_info['definition'])
+          if curr_sv_obj.mp != sv_obj.mp:
+            continue
+          if curr_sv_obj.st != sv_obj.st:
+            continue
+          if curr_sv_obj.pt != sv_obj.pt:
+            continue
+          if len(curr_sv_obj.pvs) != len(sv_obj.pvs):
+            continue
```
Comment (pradh) [inline server/lib/nl/nl_variable.py:133]: should we consider measurementDenominator too?  I think measurementQualifier being different maybe ok for peers.

File: server/lib/nl/nl_variable.py (hunk 1 above)
Comment (shifucun) [reply server/lib/nl/nl_variable.py:79]: have definition for SVG is very helpful!

File: server/lib/nl/nl_variable.py (hunk 3 above)
Comment (shifucun) [reply server/lib/nl/nl_variable.py:152]: Done

**Review by shifucun — COMMENTED** [review]
thanks for review!

File: server/lib/nl/nl_variable.py (hunk 3 above)
Comment (pradh) [reply server/lib/nl/nl_variable.py:152]: Do we need to handle md in comparison in extend_svs() ?

**Review by pradh — APPROVED** [review]
Thanks!

File: server/lib/nl/nl_variable.py (hunk 3 above)
Comment (shifucun) [reply server/lib/nl/nl_variable.py:154]: Done

**Pull request merged**